v2/git-initializer/main_test.go

```diff
@@ -0,0 +1,211 @@
+// +build integration
+
+package main
+
+import (
+	"context"
+	"log"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/brigadecore/brigade/sdk/v2"
+	"github.com/brigadecore/brigade/sdk/v2/authx"
+	"github.com/brigadecore/brigade/sdk/v2/core"
+	"github.com/brigadecore/brigade/sdk/v2/meta"
+	"github.com/brigadecore/brigade/sdk/v2/restmachinery"
+)
+
+type testcase struct {
+	name          string
+	project       core.Project
+	terminalPhase core.WorkerPhase
+}
```

Comment:
Total nit: This is only used in one place, so you can make it an anonymous type. Down on line 34 you can do:

```
testCases := []struct{
  // define fields
} {
  {
    // Case 1
  },
  ...
  {
    // Case n
  },
}
```